### ISSUE TITLE
Add support for build-time extra_hosts for Version 3

### ISSUE THREAD

**Issue by kinnalru** [docker/cli#1293]
It work well for Version 2.3 since https://github.com/docker/compose/issues/5282 but invalid for Version 3

**Comment by anibalrivero** [comment]
Is there a known workaround for this? Other than building the images outside of compose.

**Comment by kinnalru** [comment]
I'am using Version 2.3 for building images and other version for deploying.

**Comment by Stono** [comment]
Hi, 
Is there any update on this?  It feels like a real omission from v3 when it is supported by the `docker --add-host` command.

I really don't want to adjust our build process to building outside of compose.

**Comment by nicosmaris** [comment]
For the following Dockerfile:
```
FROM byrnedo/alpine-curl
```
The following works (if you replace host and IP with your case):
```
version:          '2.3'
services:
  tmp:
    build:
      context:    .
      extra_hosts:
        - "host:IP"
    command:      -kIL https://host
    tty:          true
    stdin_open:   true
```
But the following fails (if you replace host and IP with your case):
```
version:          '3'
services:
  tmp:
    build:
      context:    .
      extra_hosts:
        - "host:IP"
    command:      -kIL https://host
    tty:          true
    stdin_open:   true
```
The error is:
```
The Compose file is invalid because:                                                                                                                                                 
services.tmp.build contains unsupported option: 'extra_hosts' 
```

**Comment by posix4e** [comment]
Any updates this would be very nice

**Comment by lmammino** [comment]
Having this would be great. Currently doing weird hacks to circumvent this limitation... :(

**Comment by Maximization** [comment]
`extra_hosts` works for me in 3.7

```yml
version: "3.7"

services:
  api:
    image: playground
    container_name: api
    init: true
    ports:
      - "3000:3000"
    extra_hosts:
      - "somehost:162.242.195.82"
```

It's now a service-level key instead of nested within `build`.

**Comment by i3inc** [comment]
is this on the radar as something to be fixed?

**Comment by ghevge** [comment]
One more vote here for fixing this limitation

**Comment by thaJeztah** [comment]
This was added to the compose 3.9 schema in https://github.com/docker/cli/pull/2517. The 3.9 schema is not yet released (will be together with the next 20.x release of docker), and will have to be implemented in docker-compose as well (as the docker CLI itself doesn't use the `build` section of compose files, and only uses docker-compose files for `docker stack deploy`).

I'm closing this issue here, but subscribe / participate on https://github.com/docker/compose/issues/7323 which is a ticket in the docker-compose repository for this

**Comment by dzieciou** [comment]
It's been almost a year and half and I can see this implemented. Any update on that?

**Comment by burka** [comment]
It's solved in Version 3.9 https://github.com/docker/cli/pull/2517 - you can add `extra_hosts:` in `build:` and works for me. Maybe you need to update docker compose.
